### PR TITLE
Hotfix/sql lite

### PR DESCRIPTION
**Pull request by knowpage**

### CODE REVIEW

File: lib/common/bloc_observer/observer.dart [hunk 1]
```diff
@@ -3,19 +3,16 @@ import 'package:bloc/bloc.dart';
 class Observer extends BlocObserver{
```
Comment (ntlam6) [inline lib/common/bloc_observer/observer.dart:3]: What is this

File: lib/database/local/database.dart [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        id: maps[i]['id'],
+        popularity: maps[i]['popularity'],
+        overview: maps[i]['overview'],
+        original_title: maps[i]['original_title'],
+        poster_path: maps[i]['poster_path'],
+        title: maps[i]['title'],
+        vote_average: maps[i]['vote_average'],
+        vote_count: maps[i]['vote_count'],
+        backdrop_path: maps[i]['backdrop_path'],
+      );
```
Comment (ntlam6) [inline lib/database/local/database.dart:44]: maps.map((e) => MovieEntity(...)).toList()

File: lib/database/local/table/movie.dart [hunk 3]
```diff
@@ -0,0 +1,134 @@
+final String tableMovies = 'movies';
+
+class MovieFields {
```
Comment (ntlam6) [inline lib/database/local/table/movie.dart:3]: I think you can use enum here

File: lib/database/local/table/movie.dart [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        original_language: original_language ?? this.original_language,
+        popularity: popularity ?? this.popularity,
+        title: title ?? this.title,
+        video: video ?? this.video,
+        vote_average: vote_average ?? this.vote_average,
+        vote_count: vote_count ?? this.vote_count,
+      );
+
+  Map<String, Object?> toJson() => {
+        MovieFields.adult: adult! ? 1 : 0,
```
Comment (ntlam6) [inline lib/database/local/table/movie.dart:101]: adult == true ? 1 : 0

File: lib/database/mapper/movie_mapper/mapper_movies.dart [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        vote_average: object.vote_average,
+        vote_count: object.vote_count);
+  }
+}
+
+class MovieJsonToMovieModel implements Mapper<MovieJson, MovieModel> {
+  @override
+  MovieModel call(MovieJson object) {
+    return MovieModel(
+        id: object.id??0,
```
Comment (ntlam6) [inline lib/database/mapper/movie_mapper/mapper_movies.dart:42]: missing space : `.id ?? 0`

File: lib/presentation/journey/home/home_bloc/home_bloc.dart [hunk 6]
```diff
@@ -34,37 +43,67 @@ class HomeBloc extends Bloc<HomeEvent, HomeState> {
...
+    }
+  }
+
+  Future<void> loadFromDB() async {
+    //Load DB
+    final movieDBtoMovieModel = MovieEntityToMovieModel();
+    final listDBMovie = await dbHandler.movies();
+    for (final movie in listDBMovie) {
+      _listMovieModel.add(movieDBtoMovieModel(movie));
+    }
```
Comment (ntlam6) [inline lib/presentation/journey/home/home_bloc/home_bloc.dart:92]: can use _listMovieModel = listDBMovie.map

File: lib/presentation/journey/home/home_bloc/home_bloc.dart [hunk 7]
```diff
@@ -34,37 +43,67 @@ class HomeBloc extends Bloc<HomeEvent, HomeState> {
...
+
+  void emitMovies() {
+    if (_listMovieModel.isNotEmpty) {
+      emit(HomeDoneMovieState(listMovie: _listMovieModel, page: _page));
+    }
+  }
+
+  Future<void> loadFromDB() async {
+    //Load DB
+    final movieDBtoMovieModel = MovieEntityToMovieModel();
```
Comment (ntlam6) [inline lib/presentation/journey/home/home_bloc/home_bloc.dart:88]: DB -> Db

File: lib/presentation/journey/home/home_bloc/home_bloc.dart [hunk 8]
```diff
@@ -34,37 +43,67 @@ class HomeBloc extends Bloc<HomeEvent, HomeState> {
...
-  Future<void> loadImgOnInit() async {
-    emit(HomeLoadingImgState(listMovie: _listMovie));;
+  Future<void> loadFromAPI() async {
     final check = await client.getMovie();
     if (check.results != null) {
-      _listMovie = check.results ?? [];
-      emit(HomeDoneImgState(listMovie: _listMovie, page: _page));
+      _listMovieJson = check.results ?? [];
+      final movieJsonToModel = MovieJsonToMovieModel();
+      final movieModelToEntity = MovieModelToMovieEntity();
```
Comment (ntlam6) [inline lib/presentation/journey/home/home_bloc/home_bloc.dart:100]: Mapper should have Mapper in name

File: lib/database/local/table/movie.dart (hunk 3 above)
Comment (knowpage) [reply lib/database/local/table/movie.dart:3]: no

File: lib/database/local/table/movie.dart (hunk 4 above)
Comment (knowpage) [reply lib/database/local/table/movie.dart:101]: okay

File: lib/database/mapper/movie_mapper/mapper_movies.dart (hunk 5 above)
Comment (knowpage) [reply lib/database/mapper/movie_mapper/mapper_movies.dart:42]: okay bro

File: lib/presentation/journey/home/home_bloc/home_bloc.dart (hunk 7 above)
Comment (knowpage) [reply lib/presentation/journey/home/home_bloc/home_bloc.dart:88]: ok

File: lib/presentation/journey/home/home_bloc/home_bloc.dart (hunk 8 above)
Comment (knowpage) [reply lib/presentation/journey/home/home_bloc/home_bloc.dart:100]: ok

**Pull request merged**